ingestion-edge/ingestion_edge/publish.py

```diff
@@ -3,13 +3,50 @@
 # file, you can obtain one at http://mozilla.org/MPL/2.0/.
 
 from .conf import ROUTE_TABLE
+from .conf import QUEUE_PATH
 from datetime import datetime
 from flask import Blueprint, request
 from google.cloud.pubsub import PublisherClient
+from persistqueue import Queue
+from persistqueue.exceptions import Empty
 from typing import Tuple
+import threading
+
+FLUSH_EMPTY_STATUS = 200
+FLUSH_COMPLETE_STATUS = 200
+FLUSH_INCOMPLETE_STATUS = 504
 
 publish = Blueprint('publish', __name__)
 client = PublisherClient()
+q = Queue(QUEUE_PATH)
+flush_lock = threading.Lock()
+
+
+@publish.route("/__flush__")
+def flush_queue() -> Tuple[str, int]:
+    if q.qsize() == 0:
+        # early return on empty queue
+        return "", FLUSH_EMPTY_STATUS
+    done = 0
+    with flush_lock:
+        while True:
+            try:
+                # NOTE: this queue does not retry task if not done
+                item = q.get_nowait()
+                # NOTE: q.put writes the get state, q.task_done must be called
+                # for every get to avoid state corruption
+                q.task_done()
+            except Empty:
+                break
+            try:
+                client.publish(**item).result()
+                done += 1
+            except Exception:  # TODO only catch transient exceptions
+                # NOTE: this queue currently must requeue items on failure
+                q.put(item)
+                msg = '{"done":%d, "pending":%d}\n' % (done, q.qsize())
```

Comment:
Why do json interpolation yourself instead of using the stdlib `json`?

Reply:
*shrug* `json` is slow? i went from a non-json string to a json one and didn't think about it

Reply:
You do you, but I wouldn't hand code json using string formatting until the profiler told me this line was slow, and even then I'd be really reluctant.

Reply:
i will not do me. i will do json, like you suggest. seems good.